### DISCUSSION TITLE
Indentation issue with dropcaps following short paragraph

### DISCUSSION THREAD

**Discussion by biggnome** [sile-typesetter/sile#1742] [Q&A]
I noticed the other day that following short paragraphs starting with a drop cap, subsequent paragraphs indent as normal, e.g., this--

```
\begin[papersize=a5]{document}
\use[module=packages/dropcaps]
\use[module=packages/lorem]

\dropcap{T}his is paragraph 1. \lorem

This is paragraph 2.

\bigskip

\dropcap{P}aragraph 3 here, large and in charge.

Paragraph 4. Oh dear…\lorem

\end{document}
```
--yields this--

![dropcaptest](https://user-images.githubusercontent.com/12508219/224459821-9b971147-a805-4e77-b627-a20a7aca0bab.png)

Thoughts?

**Comment by Omikhleia** [comment]
Hello @biggnome 

Indeed, the case where a paragraph is not long enough for a dropcap is **not** handled.

It would have needed other higher level mechanisms than the one provided using hang indent/after (= the mere TeX way) to reserve the space.

As far as concerned (having played a role in the initial implementation), I decided not to bother at the time :)

But there are several considerations were typographers disagree on how such a case should be handled...
E.g.
- Should the drop cap be reduced to the available lines, here just 1? (This is what LibreOffice does, for instance).
- Or should it encompass the next paragraph, but then how is considered the paragraph indent?
- Or should it just displace the next paragraph (introducing a blank line)
 
You won't find a definitive answer (for the very few typography books where this is even mentioned), I am afraid... I might be wrong, but I think it's kind of an edge case, where drop caps are not intended to be used, i.e. the very concept of drop caps assumes the paragraph has sufficient content.

I did make some experiments a while ago, but didn't came with a satisfying solution (i.e. however one looks at it, it's not nice visually and it probably reflects bad typography, in my opinion). I might still have the code somewhere on my other computer, I will check it when possible, but I am not sure it can be made general.

**Comment by Omikhleia** [comment]
Complements:

- Another tricky consideration occurs when a paragraph skip is used between paragraphs....

- Although _technically_ not a drop cap, Bible typesetting faces this type of issue too, where the verse chapter is usually "dropped": 

  ![image](https://user-images.githubusercontent.com/18075640/224480569-4cdca68c-fff2-4b30-8675-f400f380531f.png)

  It's not uninteresting to check how short verse paragraphs are handled. Hint: it may differ between books, e.g. regular books vs. Psalms, etc.

**Comment by Omikhleia** [comment]
Since this is a "per-design" limitation, I am moving this issue to Q&A discussions. It could still be interesting to discuss and find a general approach that would be typographically sound (which can then lead to a curated issue and/or PR).

**Comment by Omikhleia** [comment]
Random finding of a case in a book where the first paragraph is too short.
![image](https://user-images.githubusercontent.com/18075640/236681151-ec43e6c6-edf9-4441-8fee-07ff26c603f9.png)

(Bunyan, _le Voyage du pèlerin_, i.e. _The Pilgrim's Progress_)

**Reply by Omikhleia** [reply]
For the record and interested readers, I experimented the above solution (propagation of the indent to the next paragraph(s) with some satisfying result:

![image](https://github.com/sile-typesetter/sile/assets/18075640/53a95bc3-ca9b-4d99-bfe0-bb2b94cdf692)

The implementation is currently in a branch of the experimental [silex.sile](https://github.com/Omikhleia/silex.sile) module. ~~I'll likely merge it to the main branch by the end of the month.~~ EDIT: It's merged.

**Reply by Omikhleia** [reply]
Oh, and with @biggnome 's own test case:

![image](https://github.com/sile-typesetter/sile/assets/18075640/59d89c34-65e7-4ed2-85d2-ada05d430278)

BTW, to be more complete on the matter, of course things don't go that well visually if there are paragraph skips...

**Comment by alerque** [comment]
I really like this. Somehow I missed your *The Pilgrim's Progress* example above whet it was posted, but that's a very elegant handling of something that might have otherwise limited the layout to a much smaller dropcap size. I could have used that on at least one project were the author was too fond of short paragraphs.
